go/vt/vttablet/tabletmanager/rpc_vreplication.go

```diff
@@ -124,6 +133,204 @@ func (tm *TabletManager) DeleteVReplicationWorkflow(ctx context.Context, req *ta
+		for i, state := range req.GetExcludeStates() {
+			if i > 0 {
+				additionalPredicates.WriteByte(',')
+			}
+			additionalPredicates.WriteString(sqltypes.EncodeStringSQL(state.String()))
+		}
+		additionalPredicates.WriteByte(')')
+	}
+	parsed := sqlparser.BuildParsedQuery(sqlReadVReplicationWorkflows, sidecar.GetIdentifier(), ":db", additionalPredicates.String())
+	stmt, err := parsed.GenerateQuery(bindVars, nil)
+	if err != nil {
+		return nil, err
+	}
+	res, err := tm.VREngine.Exec(stmt)
+	if err != nil {
+		return nil, err
+	}
+	resp := &tabletmanagerdatapb.ReadVReplicationWorkflowsResponse{}
+	if res == nil || len(res.Rows) == 0 {
+		return resp, nil
+	}
+	rows := res.Named().Rows
+	workflows := make(map[string]*tabletmanagerdatapb.ReadVReplicationWorkflowResponse, len(rows))
+
+	for _, row := range rows {
+		workflow := row["workflow"].ToString()
+		if workflows[workflow] == nil {
+			workflows[workflow] = &tabletmanagerdatapb.ReadVReplicationWorkflowResponse{Workflow: workflow}
+		}
+		workflows[workflow].Cells = row["cell"].ToString()
+		tabletTypes, inorder, err := discovery.ParseTabletTypesAndOrder(row["tablet_types"].ToString())
+		if err != nil {
+			return nil, vterrors.Wrap(err, "error parsing the tablet_types field from vreplication table record")
+		}
+		workflows[workflow].TabletTypes = tabletTypes
+		workflows[workflow].TabletSelectionPreference = tabletmanagerdatapb.TabletSelectionPreference_ANY
+		if inorder {
+			workflows[workflow].TabletSelectionPreference = tabletmanagerdatapb.TabletSelectionPreference_INORDER
+		}
+		workflows[workflow].DbName = row["db_name"].ToString()
+		workflows[workflow].Tags = row["tags"].ToString()
+		wft, err := row["workflow_type"].ToInt32()
+		if err != nil {
+			return nil, vterrors.Wrap(err, "error parsing workflow_type field from vreplication table record")
+		}
+		workflows[workflow].WorkflowType = binlogdatapb.VReplicationWorkflowType(wft)
+		wfst, err := row["workflow_sub_type"].ToInt32()
+		if err != nil {
+			return nil, vterrors.Wrap(err, "error parsing workflow_sub_type field from vreplication table record")
+		}
+		workflows[workflow].WorkflowSubType = binlogdatapb.VReplicationWorkflowSubType(wfst)
+		workflows[workflow].DeferSecondaryKeys = row["defer_secondary_keys"].ToString() == "1"
+
+		// Now the individual streams (there can be more than 1 with shard merges).
+		if workflows[workflow].Streams == nil {
+			workflows[workflow].Streams = make([]*tabletmanagerdatapb.ReadVReplicationWorkflowResponse_Stream, 0, 1)
+		}
+		stream := &tabletmanagerdatapb.ReadVReplicationWorkflowResponse_Stream{}
+		if stream.Id, err = row["id"].ToInt32(); err != nil {
+			return nil, vterrors.Wrap(err, "error parsing id field from vreplication table record")
+		}
+		srcBytes, err := row["source"].ToBytes()
+		if err != nil {
+			return nil, vterrors.Wrap(err, "error parsing binlog_source field from vreplication table record")
+		}
+		blspb := &binlogdatapb.BinlogSource{}
```

Comment:
We should just call this `bls`.  Many of the objects we use are `pb`s and it would be noisy to suffix all of them like this ....